sdk/core/scheduler/timer.h

```diff
@@ -72,6 +117,39 @@ namespace
 					break;
 				}
 			}
+			if (!Thread::any_ready())
+			{
+				for (Thread *iter = Thread::waitingList; iter;)
```

Comment:
I think I'd prefer two `waitingList`-s, one for time elapsing and one for yielding, so that we wouldn't have to do this kind of two-pass rummaging and discrimination.  I suppose with our thread counts it doesn't hugely matter, but still.

Reply:
We're only doing the second walk when we have no runnable threads, so I don't really mind that it's taking a bit more time.